modules/core/src/main/scala/sup/data/HealthReporter.scala

```diff
@@ -24,10 +24,25 @@ object HealthReporter {
   def wrapChecks[F[_]: Apply, G[_]: NonEmptyTraverse, H[_]: Reducible](checks: G[HealthCheck[F, H]])(
     implicit M: Semigroup[Health]): HealthReporter[F, G, H] = HealthCheck.liftF {
 
-    checks.nonEmptyTraverse(_.check).map { results =>
-      val status = results.reduceMap(_.value.reduce)
+    checks.nonEmptyTraverse(_.check).map(reduceResults[G, H] _)
+  }
+
+    /**
+    * Constructs a healthcheck from a non-empty structure G of healthchecks that are run in parallel.
+    * The status of the whole check is determined using the results of given checks and the semigroup of Health.
+    *
+    * e.g. if all checks need to return Healthy for the whole thing to be healthy, use [[Health.allHealthyCommutativeMonoid]].
+    */
+  def parWrapChecks[F[_]: Apply, G[_]: NonEmptyTraverse, H[_]: Reducible, M[_]](checks: G[HealthCheck[F, H]])(
+    implicit M: Semigroup[Health], P: Parallel[F, M]): HealthReporter[F, G, H] = HealthCheck.liftF {
```

Comment:
NonEmptyParallel should be enough. Additionally, if sup already has cats-par in dependencies (I think it does), use NonEmptyPar.

Reply:
using nonEmptyPar now.